test/client/fixtures/sandbox/event/message-test.js

```diff
@@ -275,13 +275,42 @@ asyncTest('timeout', function () {
 
     window.setTimeout(function () {
         ok(timeoutExceededError);
-        iframe.parentNode.removeChild(iframe);
 
         messageSandbox.PING_IFRAME_TIMEOUT = storedDelay;
 
         start();
     }, timeout);
+});
+
+asyncTest('timeout (added to DOM iframe)', function () {
+    var iframe               = document.createElement('iframe');
+    var timeoutExceededError = false;
+    var storedDelay          = messageSandbox.PING_IFRAME_TIMEOUT;
+    var timeout              = 200;
+
+    messageSandbox.PING_IFRAME_TIMEOUT = 5;
+
+    iframe.src = 'http://cross.domain.com/';
+    window.QUnitGlobals.waitForIframe(iframe)
+        .then(function () {
+            return Promise.all([
+                messageSandbox.pingIframe(iframe, 'pingCmd')
```

Comment:
It is not nested Promise.

Nested Promise (pseudo code):
```javascript
Promise
    .then(function (){
        new Promise.then(function (){
              //some code
        })
    });
```

Non nested Promise, chain of promises (my code):
```javascript
Promise
    .then(function () {
         return Promise;        
    })
    .then(function (){})
```